### PR TITLE
Fixes onnx exports for binary classification trainers

### PR DESCRIPTION
**Pull request by Lynx1820**
This PR fixes the issues with exporting the following binary classification trainers to onnx:

- SymbolicSgdLogisticRegression(),
- SgdCalibrated(),
- AveragedPerceptron(),
- FastForest(),
- LinearSvm(),
- SdcaNonCalibrated(),
- SgdNonCalibrated(),
- FastTree(),
- LbfgsLogisticRegression(),
- Trainers.LightGbm(),
- SdcaLogisticRegression(),

Note: I couldn't create consistent regression models, even when seeding, so I didn't include baseline comparison on these tests.

### CODE REVIEW

File: src/Microsoft.ML.FastTree/FastTree.cs [hunk 1]
```diff
@@ -3111,7 +3111,8 @@ bool ISingleCanSaveOnnx.SaveAsOnnx(OnnxContext ctx, string[] outputNames, string
             }
 
             string opType = "TreeEnsembleRegressor";
-            var node = ctx.CreateNode(opType, new[] { featureColumn }, outputNames, ctx.GetNodeName(opType));
+            string scoreVarName = (Utils.Size(outputNames) == 2) ? outputNames[1] : outputNames[0]; // Get Score from PredictedLabel and/or Score columns
+            var node = ctx.CreateNode(opType, new[] { featureColumn }, new[] { scoreVarName }, ctx.GetNodeName(opType));
```
Comment (harishsk) [inline src/Microsoft.ML.FastTree/FastTree.cs]: Can you please add an assert here that you expect the size of outputNames array to be at least 1 element?

File: src/Microsoft.ML.StandardTrainers/Standard/LinearModelParameters.cs [hunk 2]
```diff
@@ -240,10 +240,10 @@ JToken ISingleCanSavePfa.SaveAsPfa(BoundPfaContext ctx, JToken input)
         bool ISingleCanSaveOnnx.SaveAsOnnx(OnnxContext ctx, string[] outputs, string featureColumn)
         {
             Host.CheckValue(ctx, nameof(ctx));
-            Host.Check(Utils.Size(outputs) == 1);
-
             string opType = "LinearRegressor";
-            var node = ctx.CreateNode(opType, new[] { featureColumn }, outputs, ctx.GetNodeName(opType));
+            string scoreVarName = (Utils.Size(outputs) == 2) ? outputs[1] : outputs[0]; // Get Score from PredictedLabel and/or Score columns
+
```
Comment (harishsk) [inline src/Microsoft.ML.StandardTrainers/Standard/LinearModelParameters.cs]: An assert would be helpful here too

File: test/Microsoft.ML.Tests/OnnxConversionTest.cs [hunk 3]
```diff
@@ -187,6 +195,55 @@ public void KmeansOnnxConversionTest()
...
+                mlContext.BinaryClassification.Trainers.AveragedPerceptron(),
+                mlContext.BinaryClassification.Trainers.FastForest(),
+                mlContext.BinaryClassification.Trainers.LinearSvm(),
+                mlContext.BinaryClassification.Trainers.SdcaNonCalibrated(),
+                mlContext.BinaryClassification.Trainers.SgdNonCalibrated(),
+                mlContext.BinaryClassification.Trainers.FastTree(),
+                mlContext.BinaryClassification.Trainers.LbfgsLogisticRegression(),
+                mlContext.BinaryClassification.Trainers.LightGbm(),
+                mlContext.BinaryClassification.Trainers.SdcaLogisticRegression(),
+                mlContext.BinaryClassification.Trainers.SgdCalibrated(),
```
[the remark this answers is not in the document]
Comment (Lynx1820) [reply test/Microsoft.ML.Tests/OnnxConversionTest.cs]: oh yes, good catch.

File: src/Microsoft.ML.FastTree/FastTree.cs (hunk 1 above)
Comment (Lynx1820) [reply src/Microsoft.ML.FastTree/FastTree.cs]: sounds good.

File: src/Microsoft.ML.StandardTrainers/Standard/LinearModelParameters.cs [hunk 4]
```diff
@@ -240,10 +240,12 @@ JToken ISingleCanSavePfa.SaveAsPfa(BoundPfaContext ctx, JToken input)
         bool ISingleCanSaveOnnx.SaveAsOnnx(OnnxContext ctx, string[] outputs, string featureColumn)
         {
             Host.CheckValue(ctx, nameof(ctx));
-            Host.Check(Utils.Size(outputs) == 1);
+            Host.Assert(Utils.Size(outputs) >= 1);
```
Comment (harishsk) [inline src/Microsoft.ML.StandardTrainers/Standard/LinearModelParameters.cs]: Any particular reason for changing the runtime check to a debug check?

Comment (Lynx1820) [reply src/Microsoft.ML.StandardTrainers/Standard/LinearModelParameters.cs]: I was trying to make it consistent with the similar FastTree assert. I changed both to check.

File: src/Microsoft.ML.Data/Scorers/BinaryClassifierScorer.cs [hunk 5]
```diff
@@ -197,14 +197,32 @@ private protected override void SaveAsOnnxCore(OnnxContext ctx)
...
-                var node = ctx.CreateNode(opType, new[] { ctx.GetVariableName(outColumnNames[2]) },
-                    new[] { ctx.GetVariableName(outColumnNames[0]) }, ctx.GetNodeName(opType));
+                var node = ctx.CreateNode(opType, ctx.GetVariableName(outColumnNames[2]), binarizerOutput, ctx.GetNodeName(opType));
                 node.AddAttribute("threshold", 0.5);
+
+                opType = "Cast";
+                node = ctx.CreateNode(opType, binarizerOutput, ctx.GetVariableName(outColumnNames[0]), ctx.GetNodeName(opType), "");
+                var t = InternalDataKindExtensions.ToInternalDataKind(DataKind.Boolean).ToType();
+                node.AddAttribute("to", t);
+            }
```
Comment (harishsk) [inline src/Microsoft.ML.Data/Scorers/BinaryClassifierScorer.cs]: This section seems to be common to both the if and else block. Can this be factored out?

**Pull request merged**